### ISSUE TITLE
Never gets past pullers..

### ISSUE THREAD

**Issue by pingthings** [jupyterhub/zero-to-jupyterhub-k8s#726]
Utilizing v.06 helm chart on a K8s v1.9 cluster I get the following error in kube-controller-manager... the pullers are running but my image pulls are 1 of 2 and never progress from that in the logs of the puller pods.  This is a BARE-METAL install and not on AWS, Azure etc..

Here is the error from kube-controller-manager:

```
kube-controller-manager-pt-504 kube-controller-manager E0615 18:09:12.453866       1 daemon_controller.go:266] jhub/pre-pull-jhub-1-1529086149 failed with : error storing status for daemon set &v1.DaemonSet{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"pre-pull-jhub-1-1529086149", GenerateName:"", Namespace:"jhub", SelfLink:"/apis/apps/v1/namespaces/jhub/daemonsets/pre-pull-jhub-1-1529086149", UID:"34cf67f5-70c7-11e8-a4a2-002590491c5c", ResourceVersion:"61935290", Generation:1, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63664682952, loc:(*time.Location)(0x56358e0)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"app":"jupyterhub", "component":"image-puller", "heritage":"Tiller", "release":"jhub"}, Annotations:map[string]string{"deprecated.daemonset.template.generation":"1"}, OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.DaemonSetSpec{Selector:(*v1.LabelSelector)(0xc421adfde0), Template:v1.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"app":"jupyterhub", "component":"image-puller", "heritage":"Tiller", "release":"jhub"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container{v1.Container{Name:"image-pull-singleuser", Image:"jupyterhub/k8s-singleuser-sample:v0.6", Command:[]string{"/bin/sh", "-c", "echo \"Pulling complete\""}, Args:[]string(nil), WorkingDir:"", Ports:[]v1.ContainerPort(nil), EnvFrom:[]v1.EnvFromSource(nil), Env:[]v1.EnvVar(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList(nil)}, VolumeMounts:[]v1.VolumeMount(nil), VolumeDevices:[]v1.VolumeDevice(nil), LivenessProbe:(*v1.Probe)(nil), ReadinessProbe:(*v1.Probe)(nil), Lifecycle:(*v1.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", TerminationMessagePolicy:"File", ImagePullPolicy:"IfNotPresent", SecurityContext:(*v1.SecurityContext)(nil), Stdin:false, StdinOnce:false, TTY:false}, v1.Container{Name:"image-pull-metadata-block", Image:"jupyterhub/k8s-network-tools:v0.6", Command:[]string{"/bin/sh", "-c", "echo \"Pulling complete\""}, Args:[]string(nil), WorkingDir:"", Ports:[]v1.ContainerPort(nil), EnvFrom:[]v1.EnvFromSource(nil), Env:[]v1.EnvVar(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList(nil)}, VolumeMounts:[]v1.VolumeMount(nil), VolumeDevices:[]v1.VolumeDevice(nil), LivenessProbe:(*v1.Probe)(nil), ReadinessProbe:(*v1.Probe)(nil), Lifecycle:(*v1.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", TerminationMessagePolicy:"File", ImagePullPolicy:"IfNotPresent", SecurityContext:(*v1.SecurityContext)(nil), Stdin:false, StdinOnce:false, TTY:false}}, Containers:[]v1.Container{v1.Container{Name:"pause", Image:"gcr.io/google_containers/pause:3.0", Command:[]string(nil), Args:[]string(nil), WorkingDir:"", Ports:[]v1.ContainerPort(nil), EnvFrom:[]v1.EnvFromSource(nil), Env:[]v1.EnvVar(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList(nil)}, VolumeMounts:[]v1.VolumeMount(nil), VolumeDevices:[]v1.VolumeDevice(nil), LivenessProbe:(*v1.Probe)(nil), ReadinessProbe:(*v1.Probe)(nil), Lifecycle:(*v1.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", TerminationMessagePolicy:"File", ImagePullPolicy:"IfNotPresent", SecurityContext:(*v1.SecurityContext)(nil), Stdin:false, StdinOnce:false, TTY:false}}, RestartPolicy:"Always", TerminationGracePeriodSeconds:(*int64)(0xc421a6fdf8), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", AutomountServiceAccountToken:(*bool)(0xc421a6fee0), NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, ShareProcessNamespace:(*bool)(nil), SecurityContext:(*v1.PodSecurityContext)(0xc4241d6500), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:"", Affinity:(*v1.Affinity)(nil), SchedulerName:"default-scheduler", Tolerations:[]v1.Toleration(nil), HostAliases:[]v1.HostAlias(nil), PriorityClassName:"", Priority:(*int32)(nil), DNSConfig:(*v1.PodDNSConfig)(nil)}}, UpdateStrategy:v1.DaemonSetUpdateStrategy{Type:"RollingUpdate", RollingUpdate:(*v1.RollingUpdateDaemonSet)(0xc421d1e9a8)}, MinReadySeconds:0, RevisionHistoryLimit:(*int32)(0xc421a6fef0)}, Status:v1.DaemonSetStatus{CurrentNumberScheduled:0, NumberMisscheduled:0, DesiredNumberScheduled:0, NumberReady:0, ObservedGeneration:0, UpdatedNumberScheduled:0, NumberAvailable:0, NumberUnavailable:0, CollisionCount:(*int32)(nil), Conditions:[]v1.DaemonSetCondition(nil)}}: Operation cannot be fulfilled on daemonsets.apps "pre-pull-jhub-1-1529086149": the object has been modified; please apply your changes to the latest version and try again
```

**Comment by choldgraf** [comment]
Hello - it's going to be difficult to debug this without more information about your configuration, can you post more information on this? Also please format your error message so that it renders in a human-readable fashion. That said, bare-metal kubernetes deployments can be quite difficult, so this might also be more appropriate in one of the kubernetes repositories/chat-rooms/etc depending on what the issue is...

**Comment by consideRatio** [comment]
Hey @pingthings =)

This is a duplication of #735 #477 and fixed in #502 (Yikes we must release a final v0.7 of the chart). I recommend that you update the values you provide to helm to disable the hook pre-puller.

```yaml
# in your config.yaml somewhere, you can disable the hook pre-puller
prePuller:
  hook:
    enabled: false
```
